surf/apps/materials/views.py

```diff
@@ -355,17 +357,26 @@ def retrieve(self, request, *args, **kwargs):
 
     def create(self, request, *args, **kwargs):
         # only active and authorized users can create collection
-        check_access_to_collection(request.user)
+        try:
+            check_access_to_collection(request.user)
+        except AuthenticationFailed as exc:
+            return Response(str(exc), status=401)
         return super().create(request, *args, **kwargs)
 
     def update(self, request, *args, **kwargs):
         # only active and authorized users can update collection
-        check_access_to_collection(request.user, instance=self.get_object())
+        try:
+            check_access_to_collection(request.user, instance=self.get_object())
+        except AuthenticationFailed as exc:
+            return Response(str(exc), status=401)
         return super().update(request, *args, **kwargs)
 
     def destroy(self, request, *args, **kwargs):
         # only active and authorized users can destroy collection
-        check_access_to_collection(request.user, instance=self.get_object())
+        try:
+            check_access_to_collection(request.user, instance=self.get_object())
+        except AuthenticationFailed as exc:
+            return Response(str(exc), status=401)
```

Comment:
Whoops. Hadn't read the entire PR before commenting. Both the [update](https://github.com/encode/django-rest-framework/blob/0407a0df8a16fdac94bbd08d49143a74a88001cd/rest_framework/mixins.py#L65) and [destroy](https://github.com/encode/django-rest-framework/blob/0407a0df8a16fdac94bbd08d49143a74a88001cd/rest_framework/mixins.py#L90) methods call get_object. This is all part of the "generics API view" ecosystem. It's much easier to simply overwrite get_object and then call the super implementation of that and then run it through your method. That way you also prevent an extra roundtrip to the database (which is actually the performance bottleneck in this scenario although performance is probably fine), because get_object is not cached.